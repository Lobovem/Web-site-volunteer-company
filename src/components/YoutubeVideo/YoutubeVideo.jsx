import { TitleMain } from '../TitleMain/TitleMain';
import s from './YoutubeVideo.module.scss';

export const YoutubeVideo = () => {
  return (
    <div className={s.youtubeVideo}>
      <div className={s.youtubeVideo__wrapIframe}>
        <iframe
          className={s.youtubeVideo__iframe}
          src="https://www.youtube.com/embed/yz9zqm-TAIg"
          title="YouTube video player"
          frameBorder="0"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          allowFullScreen
          data-testid="map-iframe"
        ></iframe>
      </div>

      <TitleMain title="Допомога нашим братам та сестрам"> </TitleMain>
      <p className={s.youtubeVideo__desc}>
        Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично
        реєструє заявку, та надсилає вам на пошту підтвердження реєстрації. Для того щоб записатись на отримання допомоги, та не вистоювати
        у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження
        реєстрації.
      </p>

      <p className={s.youtubeVideo__desc}>
        За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які
        визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний
        текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове
        виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично. Нашою першочерговою метою є вирішення
        нагальних проблем ЗСУ. За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html
        -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки...
      </p>
    </div>
  );
};
