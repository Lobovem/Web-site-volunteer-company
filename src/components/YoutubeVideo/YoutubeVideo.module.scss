@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.youtubeVideo {

  &__wrapIframe {
    overflow: hidden;
    padding-bottom: 56.25%;
    position: relative;
    height: 0;
    margin-top: 60px;
    margin-bottom: 30px;
    border-radius: 12px;

    @include lg-min {
      margin-top: 100px;
    }
  }

  &__iframe {
    left: 0;
    top: 0;
    height: 100%;
    width: 100%;
    position: absolute;
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-18;
    line-height: 24px;
    color: $color-black;
    mix-blend-mode: normal;
    margin-bottom: 20px;
  }


  &__desc {
    font-weight: $font-weight-400;
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;
    margin-bottom: 20px;
  }
}