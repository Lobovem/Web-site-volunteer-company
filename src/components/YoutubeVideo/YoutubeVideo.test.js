import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { YoutubeVideo } from './YoutubeVideo';

describe('YoutubeVideo', () => {
  test('renders YoutubeVideo component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(<YoutubeVideo />);
    expect(component).toMatchSnapshot();
  });

  test('renders and search iframe element with correct attributes', () => {
    render(<YoutubeVideo />);
    const iframeElement = screen.getByTestId('map-iframe');
    expect(iframeElement).toBeInTheDocument();
  });
});
