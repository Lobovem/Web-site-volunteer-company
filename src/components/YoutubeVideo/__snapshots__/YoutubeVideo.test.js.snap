// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeVideo renders YoutubeVideo component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="youtubeVideo"
      >
        <div
          class="youtubeVideo__wrapIframe"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
            allowfullscreen=""
            class="youtubeVideo__iframe"
            data-testid="map-iframe"
            frameborder="0"
            src="https://www.youtube.com/embed/yz9zqm-TAIg"
            title="YouTube video player"
          />
        </div>
        <h2
          class="titleMain"
        >
          Допомога нашим братам та сестрам
        </h2>
        <p
          class="youtubeVideo__desc"
        >
          Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації. Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.
        </p>
        <p
          class="youtubeVideo__desc"
        >
          За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично. Нашою першочерговою метою є вирішення нагальних проблем ЗСУ. За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки...
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="youtubeVideo"
    >
      <div
        class="youtubeVideo__wrapIframe"
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          allowfullscreen=""
          class="youtubeVideo__iframe"
          data-testid="map-iframe"
          frameborder="0"
          src="https://www.youtube.com/embed/yz9zqm-TAIg"
          title="YouTube video player"
        />
      </div>
      <h2
        class="titleMain"
      >
        Допомога нашим братам та сестрам
      </h2>
      <p
        class="youtubeVideo__desc"
      >
        Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації. Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.
      </p>
      <p
        class="youtubeVideo__desc"
      >
        За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично. Нашою першочерговою метою є вирішення нагальних проблем ЗСУ. За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки...
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
